src/query/plan/rule_based_planner.hpp

```diff
@@ -567,6 +565,66 @@ class RuleBasedPlanner {
     return std::make_unique<plan::Foreach>(std::move(input_op), std::move(op), foreach->named_expression_->expression_,
                                            symbol);
   }
+
+  std::unique_ptr<LogicalOperator> GenFilters(std::unique_ptr<LogicalOperator> last_op,
+                                              const std::unordered_set<Symbol> &bound_symbols, Filters &filters,
+                                              AstStorage &storage, const SymbolTable &symbol_table) {
+    std::unique_ptr<LogicalOperator> pattern_filter =
+        ExtractPatternFilters(filters, storage, symbol_table, bound_symbols);
+    auto *filter_expr = impl::ExtractFilters(bound_symbols, filters, storage);
+    if (filter_expr) {
+      last_op = std::make_unique<Filter>(std::move(last_op), std::move(pattern_filter), filter_expr);
```

Comment:
Does this mean if user writes :

```
MATCH (n)
WHERE n.prop=1 and exists((n)-[]->(m))
RETURN n;
```

we will first try to expand and then use n.prop?

Reply:
We always try to expand first (next PR that I'm going to open).
The exists is going to perform slowly if there is a supernode with little amount of hits. However, it stops whenever it has found an edge, worst case is if it does not have an edge, then indeed it needs to go through all the edges.
However, the queries which don't have exists are not affected.

Reply:
An optimization would be to check first the expressions and then run the pattern filter, however that needs a little more architectural work, it's a bit harder to do. I'd stick to functionality for now and optimize later.

Reply:
We can measure how this will work, but in my opinion, if it will be slow this should be the first optimization point. Checking whether a node has property should be less expensive than checking for a certain pattern for a node and doing expansion.

Reply:
As we will optimize later if needed, I will resolve this.